api/management/README.md

```diff
@@ -0,0 +1,200 @@
+# Services management
+
+The API described here are intended to assist with the operation and management
+of a Veraison services deployment. They are not meant to be utilized by the
+services' users.
+
+## Policy management
+
+Policies allow augmenting attestation scheme generated [Attestation Results] on
+per-deployment basis, by specifying additional rules to be applied to these
+results. This may result in the change of the trust vector values (and,
+correspondingly, the overall result status), and/or addition of supplementary
+claims under the `policy-added-claims` EAR extension.
+
+A policy is defined by a text document describing the rules to be applied by
+the policy engine. Currently, Veraison only support the OPA policy engine.
+Correspondingly, policies must be written in Rego language used by OPA. See
+[OPA policy README] in services repo for details.
+
+Policies are automatically retrieved, based on the attestation scheme and
+tenant id for the evidence being evaluated. Currently, only one policy is
+allowed per tenant/scheme combination. If a policy is added for a tenant/scheme
+combination with an existing policy, the new policy replaces the old one. This
+also increments the internal policy version associated with the tenant/scheme
+combination.
+
+The policy management API exposes each scheme's policy via a unique resource
+path in the form `/policy/v1/{scheme}`. It supports the basic CRUD operations,
+typical to any resource management, in the following way:
+
+- **create**: A new policy is added to a scheme via a POST request to its path.
+- **read**: Policy can be retrieved via a GET request to its path.
+- **update**: A direct update is not possible, however a new policy can be
+              added and activated. Activating the new policy will deactivate
+              the old one, effectively replacing it.
+- **delete**: True deletion is not possible, in order to allow retrieval of the
+              specific policy version that was used to generate an attestation
+              result at a later date. However, all policies may be deactivated,
+              resulting in no policy being used during verification.
+
+[Attestation Results]: https://www.rfc-editor.org/rfc/internet-drafts/draft-fv-rats-ear-00.html
+[OPA Policy README]: https://github.com/veraison/services/blob/main/policy/README.opa.md#rules
+
+## REST API
+
+### Adding a new policy
+
+- A policy is added for a new scheme (with tenant implicit via auth).
+- Optionally, a name for the policy may be specified via query arguments. If a
```

Comment:
Ideally if the name is supplied as an argument then the name should also be used in the reference to the policy document, otherwise the name parameter is redundant from the user perspective!

Reply:
I did not mean removing UUID. However if we are asking user to supply a name, which in all practical purpose a user is expected to provide, then as a user I would like to retain the name (in addition to UUID) for referencing my policy in future, otherwise, what is the use for user tagging a name?

Reply:
> which in all practical purpose a user is expected to provide

In what sense? the parameter is cliearly specifiend as optional. There is no expectation, only an allowance.

> or referencing my policy in future, otherwise, what is the use for user tagging a name?

The name is not intended to reference a speicific policy (it is not unique) but to give it a meaningful label.  When a policy is retirieved, the name tells the user what the policy is for (without necessating the parsing of the rules, which, if the rule format is binary, may not even be posible without additional tools).

Having said that, it may make sense to allow filtering on name during bulk retrieval. I've added the optional argument ot the `GET /management/v1/policies/{scheme}` path.

Reply:
Sure thanks !